src/Symfony/Component/Security/Http/Attribute/IsGranted.php

```diff
@@ -39,9 +39,17 @@ public function __construct(
 
         /**
          * If set, will throw HttpKernel's HttpException with the given $statusCode.
-         * If null, Security\Core's AccessDeniedException will be used.
+         * If null and $exceptionCode given, defaults to 403.
+         * If $exceptionCode and $statusCode are null, Security\Core's AccessDeniedException will be used.
```

Comment:
This correlation looks to complex to me.
Can't we add argument $code to AccessDeniedException and decorrelate status and exception codes?

Reply:
That would make a lot of sense to me. Let me update real quick